packages/web/src/shared/accessControl.tsx

```diff
@@ -0,0 +1,131 @@
+import { subject as subjectHelper } from '@casl/ability';
+import { AbilityBuilder, createMongoAbility, PureAbility } from '@casl/ability';
+import { useQuery } from '@tanstack/react-query';
+import { ReactNode, useMemo } from 'react';
+import apiClient from '../shared/apiClient';
+import { SystemRole, LanguageRole } from '@translation/api-types';
+import queryClient from './queryClient';
+import { redirect } from 'react-router-dom';
+
+interface Actor {
+  id: string;
+  systemRoles: SystemRole[];
+  languages: {
+    code: string;
+    roles: LanguageRole[];
+  }[];
+}
```

Comment:
But now there is more information on the Actor. This data is loaded in the session